src/basic/strv.c

```diff
@@ -378,11 +378,17 @@ char *strv_join_prefix(char * const *l, const char *separator, const char *prefi
         k = strlen(separator);
         m = strlen_ptr(prefix);
 
+        if (unescape_separators) /* If there separator is multi-char, we won't know how to escape it. */
+                assert(k == 1);
```

Comment:
given that asserts in theory can be optimized away, this one is a bit weird. maybe:

 ```c
assert(!unescape_separators || k == 1);
```

Reply:
So... I know that this is the canonical form, but I find the version with separate `if` easier. The compiler doesn't care (and it should be smart enough to figure out that the conditional body is empty when compiled without asserts).